File: golang/p724.go
package main

func pivotIndex(nums []int) int {
	res := -1
	if len(nums) == 0 {
		return res
	}
	leftArr := make([]int, len(nums) + 1)
	rightArr := make([]int, len(nums) + 1)

	for i := 0; i < len(nums); i ++ {
		leftArr[i + 1] = leftArr[i] + nums[i]
		rightArr[len(nums) - i - 1] = rightArr[len(nums) - i] + nums[len(nums) - i - 1]
	}
	for i := 0; i < len(nums); i ++ {
		if leftArr[i] == rightArr[i + 1] {
			res = i
			break
		}
	}
	return res
}

File: c++/p724.cpp
class Solution {
public:
	int pivotIndex(vector<int>& nums) {
		int total = accumulate(nums.begin(), nums.end(), 0);
		int sum = 0;
		for (int i = 0; i < nums.size(); ++i) {
			total -= nums[i];
			if (total == sum) {
				return i;
			}
			sum += nums[i];
		}
		return -1;
	}
};